test/cpp/interop/client_helper.cc

```diff
@@ -88,11 +88,13 @@ std::shared_ptr<Channel> CreateChannelForTestCase(
 
   std::shared_ptr<CallCredentials> creds;
   if (test_case == "compute_engine_creds") {
-    GPR_ASSERT(FLAGS_use_tls);
+    GPR_ASSERT(FLAGS_use_tls ||
+               FLAGS_custom_credentials_type == "google_default_credentials");
     creds = GoogleComputeEngineCredentials();
     GPR_ASSERT(creds);
   } else if (test_case == "jwt_token_creds") {
-    GPR_ASSERT(FLAGS_use_tls);
+    GPR_ASSERT(FLAGS_use_tls ||
```

Comment:
Might be wrong, but it seems a little strange to run "jwt_token_token" creds test with "Google Default credentials". If I'm correct, when this test runs on a VM, "Google Default Credentials" is actually using "Compute engine creds" under the hood during this test, but the test still passes because the "jwt token creds" from the service account file contains essentially the same info as the "compute engine creds"? And if this is intended to test google default creds usage of service account file when provided with one, then does the environment variable pointing to the service account file need to be set?

Reply:
AFAIK, JWT is just a format of an Oauth token, which means a JWT token obtained via ` ServiceAccountJWTAccessCredentials` should be inter-operable with that obtained via `GoogleComputeEngineCredentials`. I am also thinking that the JWT and Oauth2 tokens used in `run_interop_tests.py` should correspond to the same service account. So, I suggest to run all tests `_AUTH_TEST_CASES` with google default credentials by passing null `creds` to `CreateTestChannel` API. After all, google default credentials should have the same contents as that of TLS + {Oauth2, JWT, Compute Engine Creds}.

Reply:
Ok, if surface interoperability between JWT token creds and google default creds is being tested, what do you think about: in the python tests runner, if the test case is "jwt_token_creds" and the creds are "google_default_creds", then set the environment variable so that "google default creds" finds the service account file and uses that. And if the test case is is anything besides "jwt_token_creds" and the creds are "google_default_creds", then leave the environment variable unset so as to be sure to use OAuth token?

Reply:
How about we do not set the environment variable if google default creds are used regardless of test cases? Since google default creds will fetch a token from metadata server if 1) env variable is not set and 2) token is not available in a pre-defined path.